소셜 로그인/인증 상태 전역 관리 및 배포환경 대응

23MinL opened the pull request:
## ⭐️ Issue Number

- #61 

## 🚩 Summary
![image](https://github.com/user-attachments/assets/c95e9487-a6a6-4445-8259-b9747ce735d6)


- 소셜 로그인(카카오/구글) 환경별 리다이렉트 URI 분기
- zustand로 로그인 상태 및 사용자 정보 전역 관리
- 헤더/TopBar/반응형에서 로그인/로그아웃 버튼 동적 표시
- /auth/success 즉시 이동 UX 개선
- 배포환경 프록시(next.config.mjs) 설정
- 로그인/회원가입/에러/대시보드 등 UI/UX 개선 및 신규 페이지/훅 추가
- README.md 등 문서화 보완

## 🛠️ Technical Concerns
- 배포환경에서 소셜 로그인 리다이렉트 및 API 프록시가 정상 동작하는지 확인 필요
- zustand 상태가 모든 레이아웃에서 일관되게 반영되는지 점검
- 신규 훅/컴포넌트/페이지가 기존 구조와 잘 통합되는지 확인
- SSR/CSR 환경에서 인증 상태 mismatch 없는지 확인

## 🙂 To Reviewer
- 로그인/로그아웃 플로우, 버튼 동작, 소셜 로그인 정상 동작 여부를 중점적으로 확인 부탁드립니다.
- 신규 추가된 훅/페이지/컴포넌트 구조에 대한 피드백도 환영합니다.
- 코드 구조나 UX 개선에 대한 의견도 부탁드립니다.

## 📋 To Do
- [ ] 배포환경에서 최종 테스트
- [ ] 코드 리뷰 반영
- [ ] merge 후 main 최신화
- [ ] .idea/ 등 불필요한 파일 .gitignore에 추가


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 소셜 로그인(카카오, 구글) 연동을 위한 로그인 플로우 및 인증 상태 관리 기능이 추가되었습니다.
  * 로그인 성공 및 실패 시 각각의 안내 페이지가 제공됩니다.
  * 로그아웃 기능이 헤더, 사이드바, 상단바 등 주요 UI에 통합되어 언제든 로그아웃이 가능합니다.
  * 프로필 카드가 실제 사용자 정보와 연동되어 동적으로 표시됩니다.
  * 사용자 인증 상태를 전역 상태로 관리하는 기능이 도입되었습니다.
  * 소셜 로그인 상태 관리를 위한 커스텀 훅과 인증 서비스가 추가되었습니다.

* **버그 수정**
  * 로그인 버튼 및 소셜 로그인 버튼의 로딩 상태와 오류 메시지 표시가 개선되었습니다.

* **문서화**
  * README의 환경 변수 예시가 수정되고, 소셜 로그인 관련 백엔드 처리 안내 문구가 추가되었습니다.

* **환경설정**
  * 환경에 따라 API 요청 경로가 자동으로 분기되도록 설정이 추가되었습니다.

* **스타일 및 UI 개선**
  * 로그인 버튼 및 소셜 로그인 버튼의 비활성화 시 시각적 피드백이 개선되었습니다.
  * 프로필 카드의 포인트 영역 레이아웃이 가로형으로 변경되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

23MinL committed:
feat: 소셜 로그인 리다이렉트 URL 환경별 분기 및 배포환경 프록시 설정

23MinL committed:
feat: zustand로 로그인 상태 및 사용자 정보 전역 관리, 헤더/TopBar 동적 로그인/로그아웃, /auth/success 즉시 이동 UX

23MinL committed:
feat: 로그인/회원가입 UI 및 훅, 에러/대시보드 페이지 등 추가 및 수정

songcw8 reviewed (approved):
안녕하세요 상민님! 소셜 로그인 플로우가 전체적으로 안정적이고 자연스럽게 잘 구현되어 있는것 같습니다. 소셜 로그인 후 마이페이지로 잘 넘어가는 흐름이 매끄럽게 작동하고 있고, 상태 관리도 깔끔하게 처리되어 있는 것 같습니다! 컴포넌트 구조도 잘 나뉘어 있어 유지보수 측면에서도 좋은 것 같습니다!
참고사항을 말씀드리자면 로그인 성공 후 사용자 정보를 불러오는 구간에 간단한 스피너 대신 스켈레톤 UI를 적용해보는건 어떨까요?? 저번에 강사님이 했던 말이 생각나서요 근데 스피너만 있어도 괜찮을 것 같습니다 어디까지나 참고사항... 고생하셨습니다!

23MinL commented:
@coderabbit review

23MinL commented on src/components/layout/top-bar.tsx:
```diff
@@ -1,40 +1,66 @@
+    } catch (error) {
+      console.error('로그아웃 실패:', error);
+    }
+  };
```
제안해주신 대로 공통 로그아웃 로직을 커스텀 훅(useAuth)으로 분리하고, 에러 처리도 개선하겠습니다.
적용 계획:
src/hooks/useAuth.ts 생성, 공통 로그아웃 함수 구현
기존 컴포넌트에서 해당 훅 사용하도록 변경

23MinL commented on src/services/auth.ts:
```diff
@@ -0,0 +1,68 @@
+    : 'http://localhost:8080/api/auth/oauth/kakao';
+  const googleRedirectUri = isProd
+    ? 'https://api.ururu.shop/api/auth/oauth/google'
+    : 'http://localhost:8080/api/auth/oauth/google';
```
보안상 환경변수로 분리하는 것이 맞다고 생각합니다. 환경변수로 수정하겠습니다.
적용 계획:
src/services/auth.ts에서 하드코딩된 값 제거, 환경변수로 대체

23MinL commented on src/services/auth.ts:
```diff
@@ -0,0 +1,68 @@
+};
+
+// 특정 제공자의 인증 URL 가져오기 (직접 OAuth URL 사용)
+export const getAuthUrl = async (provider: 'kakao' | 'google'): Promise<string> => {
```
환경 변수 미설정 시 명확한 에러 메시지를 던지도록 개선하겠습니다.
적용 계획:
getAuthUrl 함수 내 환경 변수 체크 및 예외 처리 추가

23MinL commented on src/services/auth.ts:
```diff
@@ -0,0 +1,68 @@
+      return `https://kauth.kakao.com/oauth/authorize?client_id=488328d3fc3e70249e0d7d98e1afffb7&redirect_uri=${encodeURIComponent(kakaoRedirectUri)}&response_type=code&scope=profile_nickname,profile_image,account_email&state=${state}`;
+
+    case 'google':
+      return `https://accounts.google.com/o/oauth2/v2/auth?client_id=195286873238-tks2hdhr2he1gshoj3sd89d1o5dm35m7.apps.googleusercontent.com&redirect_uri=${encodeURIComponent(googleRedirectUri)}&response_type=code&scope=openid email profile&state=${state}&access_type=offline&prompt=consent`;
```
환경 변수 미설정 시 명확한 에러 메시지를 던지도록 개선하겠습니다.
적용 계획:
getAuthUrl 함수 내 환경 변수 체크 및 예외 처리 추가

23MinL commented on src/hooks/useSocialLogin.ts:
```diff
@@ -0,0 +1,53 @@
+      setError(err.message || '구글 로그인 중 오류가 발생했습니다.');
+      setIsLoading(false);
+    }
+  }, []);
```
중복된 소셜 로그인 로직을 하나의 함수로 합쳐서 관리하겠습니다.
적용 계획:
useSocialLogin.ts에서 공통 함수로 리팩토링

23MinL commented on src/store/auth.ts:
```diff
@@ -141,4 +160,8 @@ export const useAuthStore = create<AuthState>((set) => ({
+
+  setIsLoggedIn: (loggedIn) => set({ isLoggedIn: loggedIn }),
+
+  setUserInfo: (user) => set({ userInfo: user }),
```
인증 상태 관리 로직을 일관성 있게 개선하겠습니다.
적용 계획:
src/store/auth.ts에서 인증 상태 관리 로직 개선

23MinL commented on src/hooks/useSocialLogin.ts:
```diff
@@ -0,0 +1,53 @@
+      setError(err.message || '카카오 로그인 중 오류가 발생했습니다.');
+      setIsLoading(false);
+    }
+  }, []);
```
리다이렉트 후에도 로딩 상태가 해제될 수 있도록 타임아웃을 추가했습니다.
적용: 소셜 로그인 함수 finally 블록에 setTimeout 추가.

23MinL committed:
feat: 로그아웃 로직 useAuth 훅으로 분리 및 에러 처리 개선
- 로그아웃 로직을 커스텀 훅으로 추출
- 에러 발생 시 alert로 사용자 알림 추가

23MinL committed:
refactor: 인증 상태 관리 로직 개선 및 일관성 보장
- setIsLoggedIn, setUserInfo, logout 메서드 추가
- 로그인/로그아웃 시 상태 일관성 유지

23MinL committed:
refactor: 소셜 로그인 중복 코드 제거 및 로딩 상태 관리 개선
- handleSocialLogin 함수로 중복 제거
- 리다이렉트 후 안전장치로 setTimeout 추가

23MinL committed:
fix: OAuth 클라이언트 ID/리다이렉트 URI 환경변수 적용 및 보안 강화
- 하드코딩 제거, 환경변수로 대체
- 환경 변수 미설정 시 예외 처리 추가

23MinL committed:
chore: 사용하지 않는 dashboard 페이지 파일 제거

1000hyehyang commented:
1. 비로그인 상태 시 접근 제어
[필수]
현재 장바구니, 히스토리, 주문/배송, 마이페이지 등 인증이 필요한 기능에 로그인 체크가 선제적으로 적용되어야 합니다.
→ 미로그인 시 해당 메뉴 클릭 시 로그인 페이지로 강제 이동 처리해주세요.

2. main 브랜치 pull & PR Flow
[권장]
메인 브랜치의 최근 커밋이 반영되지 않은 상태에서 PR이 발생하는 경우, 코드 충돌/리뷰 누락이 빈번히 발생합니다.
PR 전 메인 브랜치 pull 및 최신 상태 반영 후 PR 생성하시는 건 어떤가요?

3. 토스트 알림 구현 (shadcn alert/sonner)
![image](https://github.com/user-attachments/assets/a93ffa20-1601-4538-bcaf-9610e1eae4f7)
[개선]
현재 단순 div 기반 토스트를 사용 중인데,
→ shadcn/ui의 Alert 컴포넌트 또는 Sonner 등 표준화된 라이브러리 활용을 권장합니다.

4. 로딩 시 스켈레톤 UI
[UX 개선]
직접 만든 로딩 div 대신, shadcn/ui의 Skeleton 컴포넌트 적극 사용 권장.
→ 카드, 리스트, 테이블 등 각 패턴에 맞는 Skeleton 활용.
→ shadcn Skeleton 문서 및 컴포넌트별 샘플 적용 사례 참고(애니메이션/wave 등 세련된 느낌).

5. 모바일/태블릿 헤더에 로그아웃 버튼 위치
[UI/UX 관점]
'로그아웃' 버튼을 모바일/태블릿 탑 헤더에 직접 노출하는 것은 사용자 경험상 일반적이지 않습니다.
→ 보통은 사이드 메뉴(햄버거) 혹은 마이페이지/프로필 메뉴 내부에 위치시키는 것이 관례입니다. 없애는 건 어떠신가요?

6. 소셜 로그인 연동 테스트 및 백엔드 에러
[실행/테스트]
현재 백엔드에서 insert 에러가 발생하고 있어 테스트가 불가한데 요거 저만 그런가요...??

에러 상세: product_notices.customer_service_number의 VARCHAR(30) 컬럼 길이 초과

23MinL committed:
refactor: 로그아웃 로직 useLogout 커스텀 훅으로 추출 및 중복 제거 (Sidebar, MobileSidebarList, TopBar)\n- 로그아웃 로직을 useLogout 훅으로 통합\n- 토스트/라우팅/상태관리 일원화 및 UX 개선\n- 모든 로그아웃 버튼에서 동일한 방식 적용

23MinL committed:
chore: 콘솔 로그 및 민감 정보 노출 가능성 있는 디버그 코드 제거\n- 환경변수, 토큰 등 민감 정보가 노출될 수 있는 console.log/console.error 삭제\n- 필요시 개발 환경에서만 출력하도록 주석 처리

23MinL commented on src/hooks/useAuth.ts:
```diff
@@ -0,0 +1,22 @@
+      console.error('로그아웃 실패:', error);
+      alert('로그아웃 중 오류가 발생했습니다. 다시 시도해주세요.');
+    }
+  }, [logoutStore]);
```
로그아웃 로직 개선 피드백 반영 완료

- fetch 응답 상태 체크, window.location.href 대신 useRouter, alert 대신 sonner 토스트 메시지로 개선했습니다.
- 공통 로그아웃 훅(useLogout)으로 추출하여 Sidebar, MobileSidebarList, TopBar 등에서 중복 없이 재사용하도록 구조화했습니다.
- 이제 로그아웃 시 서버 응답을 확인하고, 성공 시 토스트 알림과 함께 메인으로 이동합니다.
- 콘솔 로그 등 민감 정보 노출 가능성 있는 부분도 모두 제거했습니다.

추가로 백엔드 세션/쿠키 삭제 등 추가로 체크해야 할 부분 있으면 알려줘.

23MinL commented:
피드백(접근제어, 토스트, 스켈레톤, UX 등) 모두 반영 완료했습니다.
 - /mypage 등 인증이 필요한 경로는 미들웨어에서 access_token 없으면 /login으로 강제 이동 처리
 - 토스트 알림은 sonner 라이브러리로 통일, 기존 alert/div 기반 메시지 모두 교체
 - 스켈레톤 UI는 shadcn/ui Skeleton 컴포넌트로 적용(로딩/정보 패칭 등)
 - 모바일/태블릿 UX: 헤더에서 로그인/마이페이지 버튼 제거, 로그아웃은 메뉴/사이드바에서만 노출
 - 프로필 이미지도 소셜 로그인 정보 기반으로 동작(없으면 fallback)
 - 로그아웃, 토스트, 라우팅 등 공통 로직은 커스텀 훅으로 추출해 중복 없이 관리

혹시 추가로 UX/로직적으로 더 개선이 필요한 부분 있으면 언제든 말씀해 주세요!

23MinL commented on src/components/layout/top-bar.tsx:
```diff
@@ -1,40 +1,54 @@
+              className="text-sm text-text-200 transition-colors hover:text-text-100"
             >
               마이페이지
             </Link>
```
네비게이션 인증 체크 관련 개선사항 반영

 - 장바구니, 히스토리, 주문배송, 마이페이지 등 인증이 필요한 메뉴 클릭 시
 - 비로그인 상태에서는 /login으로 강제 이동, 로그인 상태에서만 정상 이동하도록 handleProtectedLink 함수로 처리했습니다.
 - UX 관점에서 미인증 사용자의 오작동/오접근을 사전에 방지합니다.

 추가로, 서버단(middleware)에서도 /mypage는 access_token 없을 때 리다이렉트가 동작하도록 이중 보호하고 있습니다.

23MinL committed:
Merge branch 'main' into feat/61-socialLoginApi

23MinL commented on src/components/layout/header.tsx:
```diff
@@ -62,6 +79,25 @@ function MobileHeader() {
+                  로그인
+                </Button>
+              </Link>
+            )}
```
모바일/태블릿 헤더에서 로그아웃 버튼을 직접 노출하지 않고, 사이드/프로필 메뉴에서만 노출되도록 구조를 변경했습니다.
(UX 관점에서 일반적인 패턴을 따름)

23MinL commented on src/components/layout/header.tsx:
```diff
@@ -5,14 +5,31 @@ import { Search, Bell, ShoppingCart } from 'lucide-react';
+    } catch (error) {
+      console.error('로그아웃 실패:', error);
+    }
+  };
```
로그아웃 로직을 useLogout 커스텀 훅으로 통일하여 중복을 제거하고,
사용자 피드백(토스트 등)도 일관되게 개선했습니다.

23MinL committed:
chore: 중복 import 제거

23MinL committed:
feat: 소셜 로그인 API 연동 및 환경변수 설정 개선

- 소셜 로그인 시 백엔드에서 인증 URL을 받아 리다이렉트하도록 수정
- 환경변수 하드코딩 제거 및 기본값 설정
- .env.local에 NEXT_PUBLIC_API_URL 설정 추가
- 디버그용 console.log 제거
- 로컬 개발 환경에서 dev 백엔드로 요청이 정상적으로 나가도록 수정

23MinL committed:
feat: 인증, 환경변수, 레이아웃, UX 개선 및 주요 에러 수정
